src/components/HighlightedText/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import { ComponentType } from 'react';
+import { BaseProps } from '../types';
+
+export interface HighlightedText extends BaseProps {
+    HitComponent?: ComponentType;
+    TextComponent?: ComponentType;
+    className?: string;
```

Comment:
you don't need to define the `className` and `style` interface since we have defined this in `BaseProps` that you are extending.